### PR TITLE
[#40162] Update installation script to ensure successful installations

### PR DESCRIPTION
**Pull request by ghost**
[#40162] Update installation script to ensure successful installations

https://community.openproject.org/work_packages/40162

### CODE REVIEW

File: docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh [hunk 1]
```diff
@@ -5,12 +5,12 @@
 #VERBOSE LOGGING
 #set -x
 
-hash netcat 2>/dev/null
-if [ $? == 1 ]; then
-  echo
-  echo "Please install netcat (apt install netcat, yum install netcat) as this script depends on netcat. Thank you."
-  exit
-fi 
+#hash netcat 2>/dev/null
```
Comment (machisuji) [inline docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh:9]: If you don't need lines anymore, please just remove them rather than commenting them out. We can always still find them in the git history, if we should want them back.

File: docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh [hunk 2]
```diff
@@ -116,76 +114,116 @@ echo Network on localhost
...
+echo " SUMMARY "
+echo "========="
+echo
 #CHECK WEBSERVER ON LOCALHOST PORTS 80 AND 443
-echo 'Checking Port 80,443 on IP 127.0.0.1 reachable? (succeeded = reachable)'
-netcat -z -v 127.0.0.1 80 2>&1
-netcat -z -v 127.0.0.1 443 2>&1
+echo 'Checking Port 80,443 on IP 127.0.0.1 reachable? (0=YES / 1=NO)'
+#netcat -z -v 127.0.0.1 80 2>&1
+echo 2>/dev/null > /dev/tcp/127.0.0.1/80 ; echo $?
```
Comment (machisuji) [inline docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh:175]: You can use this `2>/dev/null` in line 241 too to suppress the openproject warnings while getting the version.

```
$openproject_fp run bundle exec rake version 2>/dev/null
```

I know, that's not part of this PR. But I saw you use this here, so I just wanted to point out, that it would make sense there too.

**Review by machisuji — CHANGES_REQUESTED** [review]
Looks good to me. Tested it and it works as expected. Just have one change request. Once that is addressed, this can be merged.

File: docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh (hunk 2 above)
Comment (ghost) [reply docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh:175]: there maybe is a warning when you run that command on machine where it is installed, ok, i am curious how to surpress the apt warning, maybe in the same way...

File: docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh (hunk 1 above)
Comment (ghost) [reply docs/enterprise-guide/enterprise-on-premises-guide/support/installation-support/script/op-support-data.sh:9]: I removed all hashmarked lines,  correct we have an older version in github

**Comment by ghost** [comment]
@machisuji please recheck.

**Comment by machisuji** [comment]
@adam-op It looks like you had some conflicts and didn't resolve them. I can see stuff like the following in your changes now:

```
<<<<<<< Updated upstream
#echo 'SELECT version();QUIT' | netcat $psqlserver $psqlserverport; echo $?
=======
>>>>>>> Stashed changes
echo 2>/dev/null > /dev/tcp/$psqlserver/$psqlserverport ; echo $?
```

Can you please try to resolve the conflicts? Maybe it would be easiest to just reset to the commit before (`git reset --hard 3dd0b00`) and remove the commented-out lines again. That's all the changes I requested, anyway.
